geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/AbstractRedisData.java

```diff
@@ -168,18 +168,19 @@ public void fromData(DataInput in, DeserializationContext context)
     expirationTimestamp = in.readLong();
   }
 
-  private void setDelta(DeltaInfo deltaInfo) {
-    this.deltaInfo = deltaInfo;
+  private void setDelta(DeltaInfo newDeltaInfo) {
+    deltaInfo.set(newDeltaInfo);
   }
 
   @Override
   public boolean hasDelta() {
-    return deltaInfo != null;
+    return deltaInfo.get() != null;
   }
 
   @Override
   public void toDelta(DataOutput out) throws IOException {
-    deltaInfo.serializeTo(out);
+    deltaInfo.get().serializeTo(out);
+    deltaInfo.remove();
```

Comment:
don't call remove here. It you look at storeChanges you will see it already calls setDelta(null) in a finally block. By calling ThreadLocal.remove here you just cause setDelta(null) to recreate an entry in the thread local map.
I think we should not call ThreadLocal.remove at all and the current call of setDelta(null) on storeChanges is all we need

Reply:
It's gone!